pkg/types/intoto/v0.0.1/entry.go

```diff
@@ -73,20 +73,42 @@ func NewEntry() types.EntryImpl {
 func (v V001Entry) IndexKeys() ([]string, error) {
 	var result []string
 
-	h := sha256.Sum256([]byte(v.env.Payload))
-	payloadKey := "sha256:" + hex.EncodeToString(h[:])
-	result = append(result, payloadKey)
+	// add digest over entire DSSE envelope
+	if v.IntotoObj.Content != nil && v.IntotoObj.Content.Hash != nil {
+		hashkey := strings.ToLower(fmt.Sprintf("%s:%s", swag.StringValue(v.IntotoObj.Content.Hash.Algorithm), swag.StringValue(v.IntotoObj.Content.Hash.Value)))
+		result = append(result, hashkey)
+	} else {
+		log.Logger.Error("could not find content digest to include in index keys")
+	}
+
+	// add digest over public key
+	if v.IntotoObj.PublicKey != nil && v.keyObj != nil {
```

Comment:
Resolve this:
Had to check this for myself: `keyObj` is populated in https://github.com/sigstore/rekor/blob/a166f1de5dbc1090bd27443db1f56be39d02ede0/pkg/types/intoto/v0.0.1/entry.go#L154

before `IndexKeys()` is called.